src/lib/png/png.c

```diff
@@ -0,0 +1,992 @@
+/**
+ * @file
+ * @brief lib for read and draw PNG
+ *
+ * @date May 10, 2019
+ * @author Filipp Chubukov
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/ioctl.h>
+#include <lib/png.h>
+#include <fs/file_format.h>
+
+/* 256 literals, end code, length codes + 2 unused codes */
+#define NUMBER_OF_CODE_SYMBOLS 288
+
+/* the distance codes have their own symbols, 30 used, 2 unused */
+#define NUMBER_OF_DISTANCE_SYMBOLS 32
+
+/* lengths codes */
+#define NUMBER_OF_LENGTH_CODES 19
+#define FIRST_LENGTH_INDEX 257
+#define LAST_LENGTH_INDEX 285
+
+/**
+ * the base of distances(the bits of distance
+ * codes appear after length codes and use their own huffman tree)
+ */
+const int DISTANCE_BASE[30]
```

Comment:
Please, make these variables `static` if they are not used outside of this module